WordPress/gutenberg
# .wp-env.json schema: Fix schema and add unit tests

Follow-up #63253
Implement the suggestion in [this comment](https://github.com/WordPress/gutenberg/pull/63253#pullrequestreview-2163673709)

## What?
This PR adds a unit test to validate the wp-env meta schema and the `.wp-env.json` present in the project.

At the same time, it fixes errors in the metadata detected by the tests.

## Why?

To ensure that we provide a correct meta-schema for developers.

## How?

The implementation is similar to the unit tests for other schemas.



## Testing Instructions

- Modify the [`.wp-env.json` file](https://github.com/WordPress/gutenberg/blob/trunk/.wp-env.json) and add the invalid value or property.
- `npm run test:unit test/integration/wp-env-schema.test.js `
- The `validates schema for ".wp-env.json"` test should fail.

⚠️ Note: One strange thing is that when I run this unit test _locally_, I get the following error:

```
["strict mode: missing type \"object\" for keyword \"properties\" at \"#/allOf/1\" (strictTypes)"],["strict mode: missing type \"object\" for keyword \"propertyNames\" at \"#/allOf/2\" (strictTypes)"]
```

Please let me know if this problem occurs in your environment.
